### PR TITLE
Fix `Wimplicit-fallthrough` warnings (#713)

### PR DESCRIPTION
**Pull request by AMS21**
Mostly just sprinkled some good old `[[fallthrough]]`.

### CODE REVIEW

File: src/xrPhysics/PHJoint.cpp [hunk 1]
```diff
@@ -1255,7 +1261,7 @@ void CPHJoint::SetAxisSDfactorsActive(int axis_num)
         case 0:
             dJointSetSliderParam(m_joint, dParamStopERP, axes[0].erp);
             dJointSetSliderParam(m_joint, dParamStopCFM, axes[0].cfm);
-        case 1:
+        [[fallthrough]]; case 1:
```
Comment (AMS21) [inline src/xrPhysics/PHJoint.cpp]: Is this correct? Or is this actually a missing `break`?

File: src/xrPhysics/PHJoint.cpp [hunk 2]
```diff
@@ -563,7 +563,7 @@ CPHJoint::CPHJoint(CPhysicsJoint::enumType type, CPhysicsElement* first, CPhysic
         axes.push_back(axis);
         axes.push_back(axis2);
         axes.push_back(axis3);
-    case slider: axes.push_back(axis); axes.push_back(axis);
+    [[fallthrough]]; case slider: axes.push_back(axis); axes.push_back(axis);
```
Comment (AMS21) [inline src/xrPhysics/PHJoint.cpp]: I guess this is correct but it looks very wrong to me :/

File: src/utils/xrQSlim/MxPropSlim.cpp [hunk 3]
```diff
@@ -296,8 +296,7 @@ void MxPropSlim::collect_quadrics()
         case MX_WEIGHT_AREA:
         case MX_WEIGHT_AREA_AVG:
             Q *= Q.area();
-        // no break: fallthrough
-        default:
+        [[fallthrough]]; default:
```
Comment (Xottab-DUTY) [inline src/utils/xrQSlim/MxPropSlim.cpp]: Please fix the code style for all files:
```suggestion
            [[fallthrough]];

        default:
```

File: src/xrCDB/Frustum.cpp [hunk 4]
```diff
@@ -106,40 +106,40 @@ bool CFrustum::testSphere_dirty(const Fvector& c, float r) const
     case 12:
         if (planes[11].classify(c) > r)
             return FALSE;
-    case 11:
+    [[fallthrough]]; case 11:
```
Comment (Xottab-DUTY) [inline src/xrCDB/Frustum.cpp]: 1. Code style
2. It's ok to not insert an empty line here
```suggestion
        [[fallthrough]];
    case 11:
```

File: src/xrCore/XML/tinyxmlparser.cpp [hunk 5]
```diff
@@ -100,22 +100,22 @@ void TiXmlBase::ConvertUTF32ToUTF8(unsigned long input, char* output, int* lengt
 
     output += *length;
 
-    // Scary scary fall throughs.
+    // Scary scary fallthroughs.
```
Comment (Xottab-DUTY) [inline src/xrCore/XML/tinyxmlparser.cpp]: Changing it here is not needed :)
```suggestion
    // Scary scary fall throughs.
```

File: src/xrPhysics/PHJoint.cpp
Comment (Xottab-DUTY) [inline src/xrPhysics/PHJoint.cpp:1]: @v2v3v4 what do you think about this file changes?

**Comment by AMS21** [comment]
Fixed style issues :)

**Review by Xottab-DUTY — APPROVED** [review]
Can you remove added break

File: src/xrPhysics/PHJoint.cpp
Comment (Xottab-DUTY) [reply src/xrPhysics/PHJoint.cpp:1]: Checked these changes, the are safe :)

**Pull request merged**